src/evo/mnauth.cpp

```diff
@@ -128,7 +128,10 @@ PeerMsgRet CMNAuth::ProcessMessage(CNode& peer, CConnman& connman, std::string_v
         }
     }
 
-    const uint256 myProTxHash = WITH_LOCK(activeMasternodeInfoCs, return activeMasternodeInfo.proTxHash);
+    uint256 myProTxHash;
+    if (fMasternodeMode) {
+        myProTxHash = WITH_LOCK(activeMasternodeInfoCs, return activeMasternodeInfo.proTxHash);
+    }
```

Comment:
Feels overengineered for a variable that doesn't get many opportunities to be mutated (first by  `CActiveMasternodeManager::GetProTxHash` and then by `llmq::utils::DeterministicOutboundConnection`, which already accepts a const ref).

If the value was passed through multiple functions and is a part of a large function, the tradeoff would be worth the certainty the value was never mutated but this doesn't seem like such a case.

Reply:
> Feels overengineered

only in C++ :D any other modern language would have a syntax that makes this feel more build in instead of "engineered" but the only good way to do think in c++ is to use a lambda.

```rust
let myProTxHash = if fMasternodeMode {
    WITH_LOCK(activeMasternodeInfoCs, return activeMasternodeInfo.proTxHash)
} else {
    uint256()
}
```

I guess in this instance we can do
```
const uint256 myProTxHash = fMasternodeMode ? WITH_LOCK(activeMasternodeInfoCs, return activeMasternodeInfo.proTxHash) : uint256();
```

Reply:
The latter is acceptable to me but the moment we have to add additional assertions, the former suggestion seems to be the only way to retain `const`ness.

It's been trimmed down though to be more similar to...

```c++
const uint256 myProTxHash = []() {
    if (!fMasternodeMode) return uint256();
    assert(m_mn_activeman);
    return WITH_LOCK(m_mn_activeman->cs, return m_mn_activeman->GetProTxHash());
}();
```

Resolved in latest push

Reply:
could be simplified a bit more:
```
const uint256 myProTxHash = [this]() {
    if (!fMasternodeMode) return uint256{};
    return WITH_LOCK(Assert(m_mn_activeman)->cs, return m_mn_activeman->GetProTxHash());
}();
```

anyway, this lock `cs` would be removed once https://github.com/dashpay/dash/pull/5940#discussion_r1532337139 is done

and it would become just
`const uint256 myProTxHash{fMasternodeMode ? Assert(m_mn_activeman)->GetProTxHash() : uint256{}};`

Reply:
`WITH_LOCK(Assert(m_mn_activeman)->cs, return m_mn_activeman->GetProTxHash());` doesn't work, I've mentioned this in the PR description.

> You cannot combine a `WITH_LOCK` and an `Assert` (in either mutex or accessed value), doing so will cause errors if `-Werror=thread-safety` is enabled. This is why `assert`s are added even when it would intuitively seem that `Assert` would've been more appropriate to use.